### PR TITLE
Fixed freezing GUI on reindex

### PR DESCRIPTION
**Pull request by mxaddict**
## Issue
GUI wallet freezes on start (Loading block files for reindex)

## Fix
Added a check for `fReindex` in the `BlockTipChanged` method/function when update if for a header

## Test
* Run a fully indexed node with GUI (navcoin-qt) with `-reindex=1`
* GUI wallet should not freeze when starting reindex.

### CODE REVIEW

File: src/qt/clientmodel.cpp [hunk 1]
```diff
@@ -243,7 +243,7 @@ static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int heig
         clientmodel->cachedBestHeaderTime = blockTime;
     }
     // if we are in-sync or if we notify a header update, update the UI regardless of last update time
-    if (fHeader || !initialSync || now - nLastUpdateNotification > MODEL_UPDATE_DELAY) {
+    if (fHeader && !fReindex || !initialSync || now - nLastUpdateNotification > MODEL_UPDATE_DELAY) {
```
Comment (instagibbs) [inline src/qt/clientmodel.cpp]: please put parenthesis around `fHeader && !fReindex` to make it unambiguous to readers if that's what you meant

Comment (mxaddict) [reply src/qt/clientmodel.cpp]: ACK

**Comment by mxaddict** [comment]
@instagibbs I've made the requested change 👍

File: src/qt/clientmodel.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include <net.h>
 #include <netbase.h>
 #include <util/system.h>
+#include <validation.h>
```
Comment (laanwj) [inline src/qt/clientmodel.cpp]: I don't think you're supposed to introduce a direct dependency here, either the flag should be passed in or the request should go through the node interface
@ryanofsky

Comment (mxaddict) [reply src/qt/clientmodel.cpp]: ACK, so use the node `getFreindex()` method.

Comment (ryanofsky) [reply src/qt/clientmodel.cpp]: re: https://github.com/bitcoin/bitcoin/pull/17565#discussion_r349868075

You'd replace `fReindex` here with `clientmodel->node().getReindex()`

**Comment by MarcoFalke** [comment]
Is this up for grabs?

**Comment by fanquake** [comment]
Closing as Up for grabs.

**Pull request closed without merging**